test/runner/browser/code_server.dart

```diff
@@ -4,37 +4,42 @@
 
 import 'dart:async';
 
+import 'package:http_multi_server/http_multi_server.dart';
 import 'package:shelf/shelf.dart' as shelf;
+import 'package:shelf/shelf_io.dart' as shelf_io;
+import 'package:shelf_test_handler/shelf_test_handler.dart';
 import 'package:shelf_web_socket/shelf_web_socket.dart';
-import 'package:scheduled_test/scheduled_server.dart';
 import 'package:web_socket_channel/web_socket_channel.dart';
 
-/// A class that schedules a server to serve Dart and/or JS code and receive
-/// WebSocket connections.
-///
-/// This uses [ScheduledServer] under the hood, and has similar semantics: its
-/// `handle*` methods all schedule a handler that must be hit before the
-/// schedule will continue.
+/// A class that serves Dart and/or JS code and receives WebSocket connections.
 class CodeServer {
-  /// The underlying server.
-  final ScheduledServer _server;
+  /// The handler for [_server].
+  final ShelfTestHandler _handler;
 
-  /// The URL of the server (including the port), once it's actually
-  /// instantiated.
-  Future<Uri> get url => _server.url;
+  /// The URL of the server (including the port).
+  final Uri url;
 
-  /// The port of the server, once it's actually instantiated.
-  Future<int> get port => _server.port;
+  static Future<CodeServer> start() async {
+    var server = await HttpMultiServer.loopback(0);
+    var handler = new ShelfTestHandler();
+    shelf_io.serveRequests(server, (request) {
+      if (request.method == "GET" && request.url.path == "favicon.ico") {
+        return new shelf.Response.notFound(null);
```

Comment:
[optional] Might be worth while adding a simple test favicon to keep the console free of noise.